libsolidity/CompilerContext.cpp

```diff
@@ -106,8 +106,12 @@ eth::AssemblyItem CompilerContext::getVirtualFunctionEntryLabel(FunctionDefiniti
 	solAssert(!m_inheritanceHierarchy.empty(), "No inheritance hierarchy set.");
 	for (ContractDefinition const* contract: m_inheritanceHierarchy)
 		for (ASTPointer<FunctionDefinition> const& function: contract->getDefinedFunctions())
-			if (!function->isConstructor() && function->getName() == _function.getName())
+		{
+			if (!function->isConstructor() &&
+				dynamic_cast<FunctionType const&>(*function->getType(contract)).getCanonicalSignature() ==
```

Comment:
`contract` in `getType(contract)` is the contract the function is called from, so I think this should be removed here.